includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -3816,7 +3816,14 @@ public function create_and_confirm_setup_intent() {
 		}
 
 		// Determine the customer adding the payment method, create one if we don't have one already.
-		$user        = wp_get_current_user();
+		$user = wp_get_current_user();
+		if ( ! $user->ID ) {
```

Comment:
The `create_setup_intent_ajax` method is registered to handle the `wp_ajax_create_setup_intent` action.
The `wp_ajax_*` actions are only fired for logged-in users.

Since we know that the user is logged in, in which scenario would the exception be thrown?

Reply:
Theoretically, as you mentioned, it is impossible to reach this code path without a logged-in user. This is just an additional enhancement, well, just in case in the future, this `create_and_confirm_setup_intent` method is being used somewhere else.

Reply:
🤔 Besides a bit of thinking in the future, this is more a defensive approach to ensure that this method does what it is expected to do, i.e. supporting only logged-in users.